src/parsing/pretty.ml

```diff
@@ -7,45 +7,41 @@
 
 open! Lplib
 open Base
-
 open Common
 open Console
 open Pos
 open Syntax
 open Format
+open Module
 
-let ident : ident pp = fun ff {pos; elt} ->
-  if LpLexer.is_keyword elt then
-    fatal pos "Identifier [%s] is a Lambdapi keyword." elt
-  else pp_print_string ff elt
+let ident : p_ident pp = fun ppf {pos; elt=s} ->
+  if LpLexer.is_keyword s then
+    fatal pos "Identifier [%s] is a Lambdapi keyword." s
+  else pp_print_string ppf s
 
-let arg_ident : ident option pp = fun ff idopt ->
+let arg_ident : p_ident option pp = fun ppf idopt ->
   match idopt with
-  | Some(id) -> ident ff id
-  | None     -> pp_print_string ff "_"
-
-let path_elt : Pos.popt -> (string * bool) pp = fun pos ff (s,b) ->
-  if b then fprintf ff "{|%s|}" s
-  else if LpLexer.is_keyword s then
-    fatal pos "Module path member [%s] is a Lambdapi keyword." s
-  else pp_print_string ff s
+  | Some(id) -> ident ppf id
+  | None     -> pp_print_string ppf "_"
 
-let path : Pos.popt -> p_module_path pp = fun pos ->
-  List.pp (path_elt pos) "."
+let p_mod : p_mod pp = fun ppf {elt=mp;_} -> Mod.pp ppf mp
 
-let qident : qident pp = fun ff {elt=(mp,id); pos} ->
-  let path_elt ff = fprintf ff "%a." (path_elt pos) in
-  fprintf ff "%a%a" (List.pp path_elt "") mp ident (Pos.make pos id)
+let qident : p_qident pp = fun ppf {elt=(mp,s); pos} ->
+  if LpLexer.is_keyword s then
+    fatal pos "Identifier [%s] is a Lambdapi keyword." s
+  else match mp with
+       | [] -> pp_print_string ppf s
+       | _::_ -> fprintf ppf "%a.%s" Mod.pp mp s
 
-let modifier : p_modifier pp = fun ff {elt; _} ->
+let modifier : p_modifier pp = fun ppf {elt; _} ->
```

Comment:
It would be preferable that you don't change the names of the variables if it isn't necessary, it has already been renamed from `pp` to `ff` in the past. It requires to sync back branches that are not yet merged and demands extra work.

Reply:
ppf is the name used in the code of the OCaml library itself. I don't remember why I used ff at some point.